src/plugins/saved_search/server/saved_objects/search.ts

```diff
@@ -33,44 +35,74 @@ export function getSavedSearchObjectType(
-        kibanaSavedObjectMeta: {
-          properties: {
-            searchSourceJSON: { type: 'text', index: false },
-          },
-        },
-        sort: { type: 'keyword', index: false, doc_values: false },
         title: { type: 'text' },
-        grid: { dynamic: false, properties: {} },
         version: { type: 'integer' },
-        rowHeight: { type: 'text' },
-        timeRestore: { type: 'boolean', index: false, doc_values: false },
-        timeRange: {
-          dynamic: false,
-          properties: {
-            from: { type: 'keyword', index: false, doc_values: false },
-            to: { type: 'keyword', index: false, doc_values: false },
-          },
-        },
-        refreshInterval: {
-          dynamic: false,
-          properties: {
-            pause: { type: 'boolean', index: false, doc_values: false },
-            value: { type: 'integer', index: false, doc_values: false },
-          },
-        },
-        rowsPerPage: { type: 'integer', index: false, doc_values: false },
-        breakdownField: { type: 'text' },
       },
     },
+    schemas: {
+      '8.8.0': schema.object({
+        columns: schema.arrayOf(schema.string(), { defaultValue: [] }),
+        description: schema.string({ defaultValue: '' }),
+        viewMode: schema.maybe(
+          schema.oneOf([
+            schema.literal(VIEW_MODE.DOCUMENT_LEVEL),
+            schema.literal(VIEW_MODE.AGGREGATED_LEVEL),
+          ])
+        ),
+        hideChart: schema.boolean({ defaultValue: false }),
+        isTextBasedQuery: schema.boolean({ defaultValue: false }),
+        usesAdHocDataView: schema.maybe(schema.boolean()),
+        hideAggregatedPreview: schema.maybe(schema.boolean()),
+        hits: schema.maybe(schema.number()),
+        kibanaSavedObjectMeta: schema.object({
+          searchSourceJSON: schema.string(),
+        }),
+        sort: schema.oneOf(
+          [
+            schema.arrayOf(schema.arrayOf(schema.string(), { minSize: 2, maxSize: 2 })),
+            schema.arrayOf(schema.string(), { minSize: 2, maxSize: 2 }),
+          ],
+          { defaultValue: [] }
+        ),
+        title: schema.string({ defaultValue: '' }),
+        grid: schema.object(
+          {
+            columns: schema.maybe(
+              schema.recordOf(
+                schema.string(),
+                schema.object({
+                  width: schema.maybe(schema.number()),
+                })
+              )
+            ),
+          },
+          { defaultValue: {} }
+        ),
+        version: schema.maybe(schema.number()),
+        rowHeight: schema.maybe(schema.number()),
+        timeRestore: schema.maybe(schema.boolean()),
+        timeRange: schema.maybe(
+          schema.object({
+            from: schema.string(),
+            to: schema.string(),
+            mode: schema.maybe(
+              schema.oneOf([schema.literal('absolute'), schema.literal('relative')])
+            ),
```

Comment:
We don't use it
```suggestion
```

Reply:
The problem is we use the `TimeRange` interface on our models, which means someone can send a request to save a saved search with a time range that includes `mode`, and if we don't support it in the schema, the request will fail even though our models allow it.

We can update our models to use `Omit<TimeRange, 'mode'>` instead and drop it from the schema, but that should be part of a separate PR since it's out of scope of the goal of this PR.

Reply:
I would suggest to update the model as `TimeRange` was misused there in the first place.

Reply:
I dropped `mode` from the schema, updated the `SavedSearchAttributes` model to use `Pick<TimeRange, 'from' | 'to'>` instead, and updated the mapping function for `SavedSearch -> SavedSearchAttributes` to explicitly exclude `mode`. I think this should be sufficient, but if it causes issues in CI, I'm going to revert the commit and ask that a separate issue be created for changing `timeRange`.